app/src/parser_impl.c

```diff
@@ -206,6 +225,41 @@ parser_error_t json_matchKeyValue(parsed_json_t *parsedJson,
     return parser_ok;
 }
 
+//valueTokenIdx is -1 if the optional is null
+parser_error_t json_matchOptionalKeyValue(parsed_json_t *parsedJson,
+                                  uint16_t tokenIdx, char *expectedType, jsmntype_t jsonType, uint16_t *valueTokenIdx) {
+    CHECK_PARSER_ERR(json_validateToken(parsedJson, tokenIdx))
+
+    if (tokenIdx + 4 >= parsedJson->numberOfTokens) {
```

Comment:
General remark (mostly for the audit): We should review guarantees from the json parser. I would assume that token==obj + token.size == 2 implies tokenIdx + 4 >= parsedJson->numberOfTokens. If that one is true, we don't need these additional checks. If it is false, it is definitely interesting as to why json parser could produce something with incorrect size

Reply:
This line probably should be changed, but again (I spent some time on it). together with similar lines in other functions.